extensions/simple-mcp-server/README.md

```diff
@@ -0,0 +1,179 @@
+
+### 2. Access the Server
+Once deployed, the extension provides:
+- **Web Interface**: Visit the content URL to see available tools and copy the MCP endpoint
+- **MCP Endpoint**: Located at `{direct-content-url}/mcp` for MCP client connections
+
+### 3. Use with MCP Clients
+The server can be consumed by any MCP-compatible client, including:
+- [Simple Shiny Chat](../simple-shiny-chat/README.md) extension
+- Local MCP clients
+- AI development environments that support MCP
+
+## Usage Examples
+
+### With Simple Shiny Chat Extension
+
+1. Deploy both the Simple MCP Server and Simple Shiny Chat extensions
+2. In the chat application, add the MCP server URL from this extension
+3. Ask the AI assistant to:
+   - "What datasets are available?"
+   - "Calculate summary statistics for the iris dataset"
+   - "Show me information about my Connect user account"
+
+### With Other MCP Clients
+
+Connect to the MCP endpoint at `{your-connect-server}/content/{content-guid}/mcp` and use the available tools programmatically.
+
+If you are not using the Simple Shiny Chat extension to connect to this MCP server, you will need to ensure that you can specify your Connect API key in both the `x-mcp-authorization` header and the `authorization` header for Connect API calls. Some MCP clients may not support that directly today (June 2025).
+
+## Architecture
+
+The application consists of several key components:
+
+- **FastMCP Framework**: Handles MCP protocol implementation and tool registration
+- **FastAPI Application**: Provides HTTP transport and web interface
+- **Tool Implementations**: Individual functions that implement business logic
+- **Template Engine**: Jinja2 templates for the documentation interface
+- **Dataset Storage**: In-memory storage for demonstration datasets
+
+## Tool Development
+
+### Adding New Tools
+
+To add new MCP tools, use the `@mcp.tool()` decorator:
+
+```python
+@mcp.tool()
+def your_new_tool(parameter: str) -> str:
+    """
+    Description of what your tool does.
+    
+    Args:
+        parameter: Description of the parameter
+        
+    Returns:
+        Description of the return value
+    """
+    # Your tool implementation
+    return "result"
+```
+
+### Error Handling
+
+Use `ToolError` for proper error handling:
+
+```python
+from mcp.server.fastmcp.exceptions import ToolError
+
+@mcp.tool()
+def example_tool(input_value: str) -> str:
+    if not input_value:
+        raise ToolError("Input value cannot be empty")
+    return f"Processed: {input_value}"
+```
+
+## Authentication
+
+The server supports Connect API key authentication for tools that interact with Connect services. API keys should be passed in the `x-mcp-authorization` header in the format:
+
+```
+x-mcp-authorization: Key YOUR_API_KEY
+```
+
+## Local Development
```

Comment:
I'd recommend creating a `CONTRIBUTING` for the local development notes so the `README` can be a bit more suitable for users of the extension since it linked in the Gallery.